src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -74,14 +76,22 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
+	private final String VERSION_CHARS = "[\\w\\.\\{\\}\\$]+";
```

Comment:
Since it is a constant and a regex:

```suggestion
	private static final String VERSION_CHARS_REGEX = "[\\w\\.\\{\\}\\$]+";
```

You could also compile the regex once (via `Pattern.compile( String )`) and then use it for matching.

Reply:
But then the other part is less readable, because of null-check...